src/coreclr/src/jit/lsra.cpp

```diff
@@ -6072,22 +6220,34 @@ void LinearScan::allocateRegisters()
         }
 
         // If we allocated a register, record it
-        if (currentInterval != nullptr && assignedRegister != REG_NA)
+        assert(currentInterval != nullptr);
+        if (assignedRegister != REG_NA)
         {
-            assignedRegBit                         = genRegMask(assignedRegister);
+            assignedRegBit    = genRegMask(assignedRegister);
+            regMaskTP regMask = getRegMask(assignedRegister, currentInterval->registerType);
+            // if (!RefTypeIsDef(refType))
```

Comment:
do we need this check?

Reply:
No, and I'm not sure why that was there.